### ISSUE TITLE
Feature request: Rerecording

### ISSUE THREAD

**Issue by synnae** [SourMesen/Mesen#184]
When it comes to emulation, one of the first things I check on an emulator is make sure it has rerecording. This is to help on the process of making movies and sharing gameplay footage online. 

I may not be the best at explaining, but rerecording is basically when you make a movie file to record game inputs, and you can use save states to go back to replace older inputs with newer ones. The final version of a movie will only show the newest inputs as if the old ones never happened. This is a feature present in any TAS emulator and I believe it should be staple for any decent emulator.

Was it ever considered? If so, how high priority is it?

**Comment by SourMesen** [comment]
It's in my list of things to get done eventually, along with fixing/rewriting the movie format in general to be able to properly support some things it currently doesn't (e.g VS system, FDS, rewinding, etc) and maybe have something more akin to Bizhawk or FCEUX's format (e.g a text format that can be edited by hand rather than just binary).

I can't really give you any estimate on when it might be done - probably not before I finish implementing most of the input devices that aren't emulated yet, since it makes more sense from a code point of view to do those first and ensure the input code is final/stable before I start trying to rework the movie format.  So it will probably be a while before I get to this (my free time is, unfortunately, not unlimited!)

**Comment by synnae** [comment]
I agree it'd be a good idea to rewrite the movie format so it can be edited by hand (like .fm2 or .bk2) instead of only displaying unreadable gibberish. This is pretty helpful when you want to do things like splice inputs and whatnot.

And I understand that some stuff needs to be done before you get into this. Thanks for the answer and good luck!
